engine/api/workflow/process_node.go

```diff
@@ -204,7 +204,9 @@ func processNode(ctx context.Context, db gorp.SqlExecutor, store cache.Store, pr
 		}
 	}
 	// Find an ancestor on the same repo
-	if currentRepo != "" && parentRepo != nil && parentRepo.Value != currentRepo {
+	if currentRepo != "" && parentRepo == nil {
```

Comment:
if we have something like this:    pip1-repoA  -> pip2-norepo -> pip3-repoA   we have to get information from parent.